### PR TITLE
Add lfn2pfn resolve && deletion logic.

### LINKED ISSUE #10660: MSUnmerged: Imlplement the actual deletion logic for unmerged files
**Impact of the new feature**
MSUnmerged

**Is your feature request related to a problem? Please describe.**
Once we have the proper set of unmerged files per RSE established - issue: #10657, we should implement the execution of the actual deletion command per RSE. 

**Describe the solution you'd like**
This deletion should be implemented in 3 steps:
 * Fetch the default deletion interface for the RSE from Rucio and fill it in the `MSUnmergedRSE['delInterface']` field
```
In [1]: from rucio.client import Client

In [2]: rcl = Client()

In [3]: rcl.lfns2pfns(rse='T2_US_Wisconsin',
   ...:               lfns=['cms://store/unmerged/Run2016B/JetHT/MINIAOD/ver2_HIPM_UL2016_MiniAODv2-v2/140000/388E3DEF-9F15-D04C-B582-7DD036D9DD33.root'],
   ...:               scheme='gsiftp',
   ...:               operation='delete')
Out[3]: {'cms://store/unmerged/Run2016B/JetHT/MINIAOD/ver2_HIPM_UL2016_MiniAODv2-v2/140000/388E3DEF-9F15-D04C-B582-7DD036D9DD33.root': 'gsiftp://cms-lvs-gridftp.hep.wisc.edu:2811/hdfs/store/unmerged/Run2016B/JetHT/MINIAOD/ver2_HIPM_UL2016_MiniAODv2-v2/140000/388E3DEF-9F15-D04C-B582-7DD036D9DD33.root'}

```
 * Try to aggregate as much as possible files in a single deletion execution 
 * Execute the deletions and collect counters      

**Describe alternatives you've considered**

**Additional context**

A really good example for parsing a pfn is given into the code of the Rucio client itself:
https://github.com/rucio/rucio/blob/740b825cac747280ca159089bca4635264c07fc1/lib/rucio/rse/protocols/gfal.py#L66-L187

And also for proper working with `gfal` too:
https://github.com/rucio/rucio/blob/740b825cac747280ca159089bca4635264c07fc1/lib/rucio/rse/protocols/gfal.py#L189-L205
https://github.com/rucio/rucio/blob/740b825cac747280ca159089bca4635264c07fc1/lib/rucio/rse/protocols/gfal.py#L273-L293

Here is the gfal2-python documentation:
https://dmc-docs.web.cern.ch/dmc-docs/gfal2-python.html
https://dmc-docs.web.cern.ch/dmc-docs/versions/docs/gfal2-python-epel/html/

A repository with examples:
https://gitlab.cern.ch/dmc/gfal2-bindings/-/tree/develop/example/python

And the most solid one:
https://github.com/cern-fts/gfal2-util/blob/master/src/gfal2_util/rm.py

### PR DESCRIPTION
**Pull request by todor-ivanov**
Fixes #10660 

#### Status
Ready

#### Description
With the current PR the full deletion logic for the MSUnmerged microeservice is proposed. In order to execute the set of deletions for a given RSE we follow the steps bellow:
 * Fetch the default deletion interface for the RSE from Rucio and fill it in the `MSUnmergedRSE['delInterface']` field
 * Execute the deletions by: 
    * iterating through all `lfns` to be deleted and reassembling every `pfn` by concatenating the `MSUnmergedRSE['delInterface']` + `lfn`
    * execute the `gfal rm -r pfn`
    * collect counters
  
#### Is it backward compatible (if not, which system it affects?)
YES 

#### Related PRs
https://github.com/dmwm/WMCore/pull/10677
https://github.com/cms-sw/cmsdist/pull/7148

#### External dependencies / deployment changes
Introduces a new external library dependency: `Ipython` and few more configuration parameters. The relevant PR to come soon.

We may need to mark here the external dependency from `gfal2-python`, which must already been tagged in one of the previous MSUnmerged issues.

### CODE REVIEW

File: src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py [hunk 1]
```diff
@@ -184,57 +222,164 @@ def _execute(self, rseList):
...
-            #         self.gfalCommand(rse['delInterface'], fileUnmerged)
-            #         rse['counters']['numFilesDeleted'] += 1
-            #         rse['files']['deletedSuccess'].append(fileUnmerged)
-            #     except Exception as ex:
-            #         rse['files']['deletedFail'].append(fileUnmerged)
-            #         msg = "Error while trying to delete file: %s for RSE: %s"
-            #         msg += "Will retry in the next cycle. Err: %s"
-            #         self.logger.debug(msg, fileUnmerged, rse['name'], str(ex))
-            rse['isClean'] = self._checkClean(rse)
+            ctx = gfal2.creat_context()
```
Comment (goughes) [inline src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py:235]: I don't know anything about this gfal library. Is this a typo?

File: src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py [hunk 2]
```diff
@@ -255,18 +400,18 @@ def getUnmergedFiles(self, rse):
         :param rse: The RSE to work on
         :return:    rse
         """
-        allUnmerged = self.rucioConMon.getRSEUnmerged(rse['name'])
-        while allUnmerged:
-            filePath = allUnmerged.pop()
+        rse['files']['allUnmerged'] = self.rucioConMon.getRSEUnmerged(rse['name'])
+        for filePath in rse['files']['allUnmerged']:
+            # filePath = allUnmerged.pop()
```
Comment (goughes) [inline src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py:405]: Clean up

**Review by goughes — COMMENTED** [review]
@todor-ivanov I took a quick look. My main question is about the `gfal.creat_context()` call. But, if you've tested it must not be an issue.

File: src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py (hunk 1 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py:235]: nope:) it is not a typo- it is how it is.

File: src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py (hunk 2 above)
Comment (todor-ivanov) [reply src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py]: Done. Thanks!

**Comment by todor-ivanov** [comment]
Thanks @goughes

**Review by goughes — APPROVED** [review]
Looks good. Would be great to have time for another person to test the changes, but that's not the case right now.

**Comment by amaltaro** [comment]
@todor-ivanov did you manage to fix/debug the python2 failure reported in the usual xml file:
```
    nose.failure.Failure:runTest was added with status error. Must be fixed
```
?

**Comment by todor-ivanov** [comment]
@amaltaro,  I have fixed all the tests with my unit test PR. All of them were tested  and working while running nose manually on my VM, but I could not get jenkins getting them out of stae `deleted` so that it checks them again. So at the end I merged  and was about to siscuss with you, how to get them out of this state.

**Pull request merged**